### ISSUE TITLE
magit-show-refs-popup broken for Git <2.13

### ISSUE THREAD

**Issue by dhull** [magit/magit#3203]
The magit-show-refs command when the git repository is accessed over tramp has become broken over the last few days. If I open a magit-status buffer over tramp, and then run `magit-show-refs-popup` from within that status buffer, the resulting `*magit-refs` buffer doesn't show the branches.

```elisp
(magit-status "/ssh:david.hull@cos6:src/ssrtb-server/")
```

Now:
```
Branches:

Origin (git@example.com:orange/ssrtb-server.git):

Tags (385)
  rpm-9_9_0-x86_64          tag 'rpm-9_9_0-x86_64' created by fw-revision-git
  rpm-9_99_0-x86_64         tag 'rpm-9_99_0-x86_64' created by fw-revision-git
```

Previously (as of around Tuesday):
```
Branches (7)
  apps-15392-device-id-throttle-priority WIP.
@ apps-15447-map-request-state Update for ox-request-context-6.0.0.  Version 9.172.0.
  expose-endpoint-latency-via-http-and-json ssrtb_wm_latency: New module to report latency stats over HTTP.
  jsons                     oxjson: Pre-convert object keys.
  master                    [origin/master] APPS-15451: Initialize token bucket to full capacity.  Version 9.171.2.
  resolicit-2.5-web-index   SOLICIT-133: Update web_index to call ssrtb_service:solicitBidders.
  set-per-bidder-worker-heap-size Set initial heap size when spawning per-bidder workers.

Origin (git@example.com:orange/ssrtb-server.git) (2)
  origin/buy-690-notify-error-reply BUY-413: Hide private fields for external users.
  origin/master             APPS-15451: Initialize token bucket to full capacity.  Version 9.171.2.

Tags (385)
  rpm-9_9_0-x86_64          tag 'rpm-9_9_0-x86_64' created by fw-revision-git
  rpm-9_99_0-x86_64         tag 'rpm-9_99_0-x86_64' created by fw-revision-git

```

Magit version: `Magit 20171011.1130, Git 2.7.4, Emacs 25.3.2, darwin`

**Comment by dhull** [comment]
When I revert magit to 8b3099c0afd8548d4bff8d38ad8229a3d1610d49 the problem does not occur.

**Comment by tarsius** [comment]
What `git` version are you using on the remote machine?

**Comment by dhull** [comment]
This is the remote machine I am accessing over tramp:
```shell
$ git --version
git version 2.7.4
```
And this is the local machine that emacs is running on:
```shell
$ git --version
git version 2.13.5 (Apple Git-94)
```

**Comment by dhull** [comment]
It appears that I was jumping to conclusions when I blamed the problem on tramp. I see the same output when I run emacs directly on the remote machine.

Magit version: `Magit 20171011.1130, Git 2.7.4, Emacs 25.1.1, gnu/linux`

**Comment by MatthewFluet** [comment]
I ran into this as well and determined that it was due to git version 2.7.4 (as provided by Ubuntu 16.04LTS) not understanding `branch --format=...` (#3198).

**Comment by kyleam** [comment]
My guess is that the call to Git is throwing an error because we're
passing a flag that version 2.7.4 doesn't support.  The minimum Git
for Magit is 1.9.4.

**Comment by kyleam** [comment]
Looks like `--format` was added in Git 2.13.0 (3d9e4ce3e).

**Comment by wombatzus** [comment]
This is still broken for me with local branches, seems like there is a ```--format``` parameter that is still throwing errors: 
```
$ git "for-each-ref" "--format=\
%(HEAD)%00%(refname:short)%00%(objectname:short)%00%(subject)%00\
%(upstream:short)%00%(upstream)%00%(upstream:track,nobracket)"
fatal: unrecognized format: %(upstream:track,nobracket)
$ git --version 
git version 2.11.0
```

**Comment by kyleam** [comment]
> This is still broken for me with local branches

Alright, thanks for the info.  Seems like "upstream:track,nobracket"
was also added in 2.13 (7743fcca5).  I'll have another look and will
make sure to test with the minimum Git version this time.

**Comment by kyleam** [comment]
> I'll have another look

fixed with 5c7ad0293

> and will make sure to test with the minimum Git version this time.

or not---I'm having issues building v1.9.4.  But I did test for
compatibility with v2.10, and I looked at v1.9.4's manpage for "git
for-each-ref" and it seems like all the fields we use are supported.

**Comment by wombatzus** [comment]
Thank you, working for me now. Much appreciated!
